internal/server/middleware.go

```diff
@@ -304,3 +290,33 @@ hostLookup:
 	}
 	return org, nil
 }
+
+func reqBearerToken(req *http.Request) (string, error) {
+	header := req.Header.Get("Authorization")
+
+	bearer := ""
+
+	parts := strings.Split(header, " ")
+	if len(parts) == 2 && parts[0] == "Bearer" {
+		bearer = parts[1]
+	} else {
+		// Fall back to checking cookies
+		cookie, err := getCookie(req, cookieAuthorizationName)
+		if err != nil {
+			logging.L.Trace().Msg("auth cookie not found, falling back to signup cookie")
+			cookie, err = getCookie(req, cookieSignupName)
+			if err != nil {
+				return "", fmt.Errorf("%w: valid token not found in request", internal.ErrUnauthorized)
+			}
+		}
```

Comment:
I think the only place we'd need to call `exchangeSignupCookieForSession` would be here, right? 

We've looked for the regular `auth` cookie, didn't find it, but we did find the signup cookie. So we could do the exchange at this point.

Although I'm not sure what happens if the first request returns a non-2xx code. Does the browser still save the cookie? If not, then I think the way you have it now is much safer.

Reply:
> Origin servers MAY send a Set-Cookie response header with any response. User agents MAY ignore Set-Cookie headers contained in responses with 100-level status codes but MUST process Set-Cookie headers contained in other responses (including responses with 400- and 500-level status codes). An origin server can include multiple Set-Cookie header fields in a single response. The presence of a Cookie or a Set-Cookie header field does not preclude HTTP caches from storing and reusing a response.

Looks like it should still be set, I'll move it here and test it